pkg/server/endpoints/node/handler.go

```diff
@@ -53,17 +55,31 @@ type Handler struct {
 	c       HandlerConfig
 	limiter Limiter
 
-	dsCache *datastoreCache
+	dsCache                          *datastoreCache
+	fetchRegistrationEntriesCache    *regentryutil.FetchRegistrationEntriesCache
+	useFetchRegistrationEntriesCache bool
```

Comment:
We can probably do without this flag and just check if the cache is non-nil. Doing it that way will prevent possibly invalid states, like this flag being true and the cache being nil, and vice-versa.